### ISSUE TITLE
Change shell to real shell for ShellCheck

### ISSUE THREAD

**Issue by zemanlx** [hadolint/hadolint#226]
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### What

We should also pass a value from `SHELL` to `shellcheck` in 
https://github.com/hadolint/hadolint/blob/7de84c2386b38234e9f6a01dd308b431d4404260/src/Hadolint/Bash.hs#L26 because current setting is hiding errors if shell is not `bash` which is mostly not true because [default is `sh`](https://docs.docker.com/engine/reference/builder/#shell).

### Steps to reproduce the behaviour

```Dockerfile
# Dockerfile.shell
FROM ubuntu:18.04 AS builder
SHELL ["/bin/bash", "-c"]
RUN echo $RANDOM
SHELL ["/bin/sh", "-c"]
RUN echo $RANDOM
```

```bash
$ docker build -t shelly -f Dockerfile.shell .
Sending build context to Docker daemon  2.953MB
Step 1/5 : FROM ubuntu:18.04 AS builder
 ---> 113a43faa138
Step 2/5 : SHELL ["/bin/bash", "-c"]
 ---> Running in 2689b9e0803c
Removing intermediate container 2689b9e0803c
 ---> f28dd8e83d5b
Step 3/5 : RUN echo $RANDOM
 ---> Running in 60458bd71409
13845
Removing intermediate container 60458bd71409
 ---> 33b682baec24
Step 4/5 : SHELL ["/bin/sh", "-c"]
 ---> Running in e1c238a26414
Removing intermediate container e1c238a26414
 ---> b0792e75154e
Step 5/5 : RUN echo $RANDOM
 ---> Running in 94dfc474759a

Removing intermediate container 94dfc474759a
 ---> a3e24dd32283
Successfully built a3e24dd32283
Successfully tagged shelly:latest

# Notice that $RANDOM is undefined in sh

$ hadolint Dockerfile.shell 
$
$ cat >shelly.sh                       
echo "$RANDOM"

# check script for /bin/sh
$ shellcheck -s sh shelly.sh           

In shelly.sh line 1:
echo "$RANDOM"
      ^-- SC2039: In POSIX sh, RANDOM is undefined.

# check script for /bin/bash
$ shellcheck -s bash shelly.sh
$
```

**Comment by lorenzo** [comment]
great idea!
